src/tests/test_courses.py

```diff
@@ -196,21 +198,28 @@ def test_lessons_appear_on_endpoint(self, user_client):
         assert response.json()["course_id"] == lesson.chapter.course_id
 
     def test_lessons_pagination_works(self, user_client):
-        """Тест, что работает переключение между уроками."""
-        chapter = ChapterFactory()
-        lesson_1 = LessonFactory(chapter=chapter, order_number=1)
-        lesson_2 = LessonFactory(chapter=chapter, order_number=2)
-        lesson_3 = LessonFactory(chapter=chapter, order_number=3)
-        url_1 = reverse("lessons-detail", kwargs={"pk": lesson_2.id})
-        url_2 = reverse("lessons-detail", kwargs={"pk": lesson_3.id})
-        response_1 = user_client.get(url_1)
-        response_2 = user_client.get(url_2)
-        assert response_1.status_code == status.HTTP_200_OK
-        assert response_2.status_code == status.HTTP_200_OK
-        assert response_1.json()["next_lesson_id"] == lesson_3.id
-        assert response_1.json()["prev_lesson_id"] == lesson_1.id
-        assert response_2.json()["next_lesson_id"] is None
-        assert response_2.json()["prev_lesson_id"] == lesson_2.id
+        """
+        Тест, что работает переключение между уроками.
+
+        Создается курс с 2 Главами по 4 урока в каждой.
+        Проверяется навигация по всем 8 урокам, при этой первый и последний урок
+        должны выдавать None при отсутствии крайних уроков.
+        """
+        _ = CourseWith2Chapters()
+        lessons = Lesson.objects.all().order_by("id")
+        for number, lesson in enumerate(lessons):
+            url = reverse("lessons-detail", kwargs={"pk": lesson.id})
+            response = user_client.get(url)
+            assert response.status_code == status.HTTP_200_OK
+            if number == 0:
+                assert response.json()["prev_lesson_id"] is None
+                assert response.json()["next_lesson_id"] == lesson.id + 1
+            elif number == 7:
+                assert response.json()["prev_lesson_id"] == lesson.id - 1
+                assert response.json()["next_lesson_id"] is None
+            else:
+                assert response.json()["prev_lesson_id"] == lesson.id - 1
+                assert response.json()["next_lesson_id"] == lesson.id + 1
```

Comment:
то что id + 1 это не факт

измени порядок обхода, сделай обход по коллекции:


```suggestion
        lessons = Lesson.objects.all().order_by("id")
        prev_lesson = None
        for number, lesson in enumerate(lessons):
            url = reverse("lessons-detail", kwargs={"pk": lesson.id})
            response = user_client.get(url)
            assert response.status_code == status.HTTP_200_OK
            assert response.json()["prev_lesson_id"] = prev_lesson
            prev_lesson = response.json()["prev_lesson_id"]


        lessons = Lesson.objects.all().order_by("-id")
        next_lesson = None
        for number, lesson in enumerate(lessons):
            url = reverse("lessons-detail", kwargs={"pk": lesson.id})
            response = user_client.get(url)
            assert response.status_code == status.HTTP_200_OK
            assert response.json()["next_lesson_id"] = next_lesson
            next_lesson = response.json()["next_lesson_id"]
```

Соответственно через параметры избавиться от дублирования: достаточно два параметра: ключ сортировки, ключ из json